quamash/_windows.py

```diff
@@ -15,22 +15,23 @@
 
 import math
 
-from . import QtCore
+from . import _make_signaller
 from ._common import with_logger
 
 UINT32_MAX = 0xffffffff
 
 
-class _ProactorEventLoop(QtCore.QObject, asyncio.ProactorEventLoop):
+class _ProactorEventLoop(asyncio.ProactorEventLoop):
 
 	"""Proactor based event loop."""
 
 	def __init__(self):
-		QtCore.QObject.__init__(self)
-		asyncio.ProactorEventLoop.__init__(self, _IocpProactor())
+		super().__init__(_IocpProactor())
 
-		self.__event_poller = _EventPoller()
-		self.__event_poller.sig_events.connect(self._process_events)
+		self.__event_signaller = _make_signaller(self._qtcore, list)
```

Comment:
I see where you're coming from, but the IOCP Loop for windows and the
Selector Loop for unix are super-classes because they need to be.

It seems like you're saying QEventLoop should pass qtcore up to it's parent
(which will be different depending on platform) and then IOCP Loop or
Selector loop class should set it's own _qtcore? then QEventLoop inherits
that _qtcore.

That feels like *more* spaghetti, not less. Maybe if there was a shared
baseclass, but that would mean adding a new class for one protected
variable.

OTOH it does violate some OOP principle (and laws of nature) that a parent
can't inherit properties from it's children.

Anyway after some consideration I've made those changes.


On Mon, Jan 19, 2015 at 12:01 PM, Jonas Wielicki <notifications@github.com>
wrote:

> In quamash/_windows.py
> <https://github.com/harvimt/quamash/pull/29#discussion-diff-23183138>:
>
> >
> > -		self.__event_poller = _EventPoller()
> > -		self.__event_poller.sig_events.connect(self._process_events)
> > +		self.__event_signaller = _make_signaller(self._qtcore, list)
>
> I personally would go for passing it through super() and let the base
> class deal with setting the _qtcore attribute. Make it part of the
> ("protected", in C++ terms) interface a base class for QEventLoop must
> implement.
>
> (Or just document it.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/harvimt/quamash/pull/29/files#r23183138>.
>